File: backend/FactoryOps.Api/Database/Models/Item.cs
namespace FactoryOps.Api.Database.Models;

public class Item : BaseEntity
{
	public int Group { get; set; }
	public string Title { get; set; } = string.Empty;
	public DateTime StartTime { get; set; }
	public int Length { get; set; }
	public bool CanMove { get; set; }
	public bool CanResize { get; set; }
	public bool CanChangeGroup { get; set; }
}

File: frontend/factory-ops/src/models/Item.ts
import { Programmer } from './Programmer';

export interface Item {
	id: number;
	group: number;
	title: string;
	start_time: number;
	end_time: number;
	length: number;
	canMove: boolean;
	canResize: boolean;
	canChangeGroup: boolean;
	programmer?: Programmer;
}
